report/src/main/java/com/hartwig/actin/report/pdf/tables/molecular/RecentMolecularSummaryGenerator.java

```diff
@@ -1,148 +1,186 @@
-        }
+    private Cell genesWithHighDriverMutationCell() {
+        return summaryCellForGeneAlterations(molecular.drivers().variants().stream());
     }
 
     @NotNull
-    private static Cell addIndent(@NotNull Cell cell) {
-        return cell.setPaddingLeft(10);
+    private Cell genesWithHighDriverCopyNumberCell(boolean isGain) {
+        return summaryCellForGeneAlterations(molecular.drivers()
+                .copyNumbers()
+                .stream()
+                .filter(copyNumber -> copyNumber.type().isGain() == isGain));
     }
 
     @NotNull
-    private static String formatTreatmentsPerEvent(@NotNull Multimap<String, String> treatmentsPerEvent) {
-        Set<String> evidenceStrings = Sets.newTreeSet();
-        for (Map.Entry<String, String> entry : treatmentsPerEvent.entries()) {
-            evidenceStrings.add(entry.getKey() + ": " + entry.getValue());
-        }
-        return concat(evidenceStrings);
+    private Cell genesWithHighDriverHomozygousDisruptionCell() {
+        return summaryCellForGeneAlterations(molecular.drivers().homozygousDisruptions().stream());
     }
 
     @NotNull
-    private static String concat(@NotNull Set<String> strings) {
-        StringJoiner joiner = Formats.commaJoiner();
-        for (String string : strings) {
-            joiner.add(string);
-        }
-        return Formats.valueOrDefault(joiner.toString(), "None");
+    private Cell highDriverGeneFusionsCell() {
+        String fusions = molecular.drivers()
+                .fusions()
+                .stream()
+                .filter(fusion -> fusion.driverLikelihood() == DriverLikelihood.HIGH)
+                .map(Driver::event)
+                .collect(Collectors.joining(", "));
+        return Cells.createValue(fusions);
+    }
+
+    @NotNull
+    private Cell highDriverVirusDetectionsCell() {
+        String fusions = molecular.drivers()
+                .viruses()
+                .stream()
+                .filter(virus -> virus.driverLikelihood() == DriverLikelihood.HIGH)
+                .map(virus -> String.format("%s (%s integrations detected)", virus.event(), virus.integrations()))
+                .collect(Collectors.joining(", "));
+        return Cells.createValue(fusions);
+    }
+
+    @NotNull
+    private Cell actionableGenesWithMedOrLowDriverMutation() {
+        Set<DriverLikelihood> allowedLikelihoods = Set.of(DriverLikelihood.LOW, DriverLikelihood.MEDIUM);
+
+        Set<String> eventsWithActinTrials = trials.stream()
+                .filter(EvaluatedTrial::isPotentiallyEligible)
+                .filter(EvaluatedTrial::isOpen)
+                .flatMap(trial -> trial.molecularEvents().stream())
+                .collect(Collectors.toSet());
+
+        String genes = Stream.concat(molecular.drivers().variants().stream(), molecular.drivers().disruptions().stream())
+                .filter(driver -> driver.driverLikelihood() != null)
```

Comment:
I added this line since without this the code would give a NPE on production samples. Not sure why though? "if (set.contains(null))" is valid code I thought?

Reply:
I would prefer to use explicit null check rather than guava. The explicit null checks we can clean up as we move towards optionals, while guava may hide unnoticed for years :)

Reply:
One functional comment about this code, I think you can add every type of driver to this stream, not just the variants and disruptions. Or is there a specific reason why you add only these 2?

Reply:
I just added those 2 based on a literal interpretation of Nina's comment:
> i.e. medium/low driver mutation or disruption

I don't know what the clinician is trying to see here, but I can add other driver types as well if there's value. This implementation is limited to Driver implementations that also implement `GeneAlteration` to retrieve the specific gene. That means we could easily add copy number variations and homozygous disruptions, but would need to return a different string if we wanted to include fusions or viruses. Should I include all gene alterations?

Regarding the non-drivers, it sounds like what we actually want to include here is all entries with likelihood != HIGH. That takes care of the null check too :)

Reply:
It would be cleanest to include all gene alterations and fusions. LOW driver fusions could in theory be actionable just like LOW driver variants. 

And yes, != HIGH should do the trick!

Reply:
I just discussed this with Nina and we agreed on some improvements. The new reported groups would include:
- Genes with high driver mutation: variant - HIGH
- Amplified genes: copy number gains - all likelihoods (should always be HIGH)
- Deleted genes: copy number losses - all likelihoods (should always be HIGH)
- Homozygously disrupted genes - all likelihoods
- Gene fusions - all likelihoods
- Virus detection - all likelihoods
- Actionable genes with medium/low driver: variant with likelihood != HIGH and disruptions (should never be HIGH)

This should include all drivers of interest. Do you agree?

Reply:
The report should not make assumptions about the fact that disruptions are never HIGH drivers. 

Also, there may be amplified genes with driverLikelihood == null, would we want to display those? 

I personally think your original proposal with my proposed adjustments is cleaner. It makes no assumptions about what events could have what driver likelihoods and also is 100% exhaustive. We never miss a HIGH or actionable driver of any kind, without making assumptions about actionability.

Reply:
Nina is sitting next to me and understood the route that we are on, but noted that "actionable genes" should be renamed to "actionable events", since viruses/fusions are not genes. Also we may want to display the events rather the genes, would need to test if this doesn't lead to super long lists of events for some samples.

Reply:
I agree that we shouldn't assume any likelihoods and risk missing useful data. If we're going to continue limiting each grouping to high likelihood, should we change the row titles to add " with high driver likelihood"?

Actionable events (vs. genes) is a good change if we want to extend this beyond gene alterations, though the event representations of the gene alterations are definitely longer. Do we want to include viruses here and limit the earlier virus row to HIGH? Should we limit the fusions and homozygous disruptions rows to HIGH as well? Should we add a row for HIGH disruptions?

As a space-saving measure, we could consider hiding some of these rows when they're empty if they aren't usually relevant.

Reply:
I think we can leave the rows as they are. Very strictly speaking we assume that disruptions are never HIGH but I think that's fine. Conditionally showing rows is dubious since "none" is a meaningful result, so better always show the same columns. I can do a review of the final column names once the logic is completed. I think its quite subtle, and we generally don't want to push the concept of driver likelihood to end users

Reply:
I added a row for HIGH disruptions but I can take it out if it's just noise. Rather than making an assumption about likelihood, I could include all disruptions in the actionable events field and only apply likelihood filtering to the other drivers. Would that be the ideal solution?

I agree it's better to show "None" when we have that information.

Reply:
"Would that be the ideal solution" -> Yes it sounds good to me!